app/src/main/java/net/tuxed/vpnconfigimporter/MainActivity.java

```diff
@@ -76,7 +72,9 @@ protected void onNewIntent(Intent intent) {
         }
         try {
             _connectionService.parseCallbackIntent(intent);
-            openFragment(new ConnectProfileFragment());
+            // Remove it so we dont parse it again.
+            intent.setData(null);
```

Comment:
Hmm I will retest this. Problem was that URI was still there when the configuration changed (recreated activity), and the URL would be parsed twice.